### DISCUSSION TITLE
Would Peripherals get UUID support?

### DISCUSSION THREAD

**Discussion by DariusSG** [cc-tweaked/CC-Tweaked#891] [Q&A] [answered]
Similar to OpenComputers Component addressing system, would this be considered to be added and why/why not? 

Note: Peripherals also get persistent UUIDs especially for blocks connected via wired modems. (attach a chest via wired modems and run peripherals program. replace the wired modem and it will change the chest peripheral name to _1) It gets annoying especially if you had to replace all the wired modems, Programs that relied on hardcoded peripherals' names will break.

**Comment by SquidDev** [comment, marked as answer]
Probably not, no. There's a couple of reasons for this:
 - The public API doesn't really allow for this. Unlike OC, peripherals don't have any state, so there's nowhere to store this UUID.
 - CC has a pretty different mental model for peripherals than OC does, and I don't think addressing really fits in to it.

> It gets annoying especially if you had to replace all the wired modems

Any reason you have to do this? This sounds like a more serious problem.

> Programs that relied on hardcoded peripherals' names will break.

Any solution is always going to have this problem - the same happens if you need to break/move a peripheral.

---

One possibility, which was vaguely discussed in #78 and #703 was to allow nametags to specify custom additional names/labels. Still not really sure how I feel about it though.

**Reply by DariusSG** [reply]
Well, thanks for clarifying. Thanks for bringing up custom names/labels via nametags. It's a good idea. I might consider adding it to my custom build of CC: Tweaked. 

Question, Are you going to add anything from CCTweaks`https://github.com/SquidDev-CC/CCTweaks/`, Specifically the turtles able to refuel with RF and also what are your thoughts on adding groovy scripts alongside Lua for performance increase and for really advanced programs/code, or will it be a security issue/not feasible for other reasons such as overhead cost.

**Reply by SquidDev** [reply]
> Are you going to add anything from CCTweaks

Nothing which hasn't been ported across already, no. I think most of the remaining features aren't bad ideas, but just don't really fit in the base mod.

As far as RF goes, the API is there, so other mods are free to register a fuel provider (which is what [Plethora](https://plethora.madefor.cc/) did in 1.12). I just don't want to anywhere near the gnarly world of RF/FE.

> Adding groovy scripts alongside Lua for performance increase

Afraid not. https://github.com/SquidDev-CC/CC-Tweaked/issues/547#issuecomment-701335369 (and #769 to a lsesser extent) touch on this a bit, but Lua touches a sweet spot of "good enough" while also being easy to sandbox (and maintain).

**Reply by SirEdvin** [reply]
> Specifically the turtles able to refuel with RF

You can do this with Advanced Peripherals, but only for specific type of turtle (with automata cores)
